vmdb/app/models/ems_kubernetes.rb

```diff
@@ -11,19 +11,40 @@ def self.description
     @description ||= "Kubernetes".freeze
   end
 
-  def self.raw_connect(hostname, port, api_version)
+  def self.raw_connect(uri, api_version)
```

Comment:
Agreed.  The URI should be built by us to ensure IPv6 and other things are considered.